File: Day-01/python/part2.py
data_in = open("../data.txt",'r')
data_first :list[int] = []
data_second:list[int] = []
for line in data_in:
    x,y = line.split('   ')
    x,y = int(x),int(y)
    data_first.append(x)
    data_second.append(y)

sim_score = 0
for x in data_first:
    sim_score+=x*data_second.count(x)

print(sim_score)

File: Day-01/rust/src/part2.rs
// part2.rs

pub fn solve(input_path: &str) -> Result<i32, Box<dyn std::error::Error>> {
    let data_pairs: Vec<(i32, i32)> = std::fs::read_to_string(input_path)?
        .lines()
        .map(|line| {
            let mut nums = line.split("   ");
            (nums.next().unwrap().parse().unwrap(), 
             nums.next().unwrap().parse().unwrap())
        })
        .collect();
    let data_first: Vec<i32> = data_pairs.iter()
        .map(|(x,_)| *x)
        .collect::<Vec<_>>();
    let data_second: Vec<i32> = data_pairs.iter()
        .map(|(_,y)| *y)
        .collect::<Vec<_>>();

    let sim_score: i32 = data_first.iter()
        .map(|&x|x * data_second.iter().filter(|&y| x == *y).count() as i32)
        .sum();
    
    // Rest of solution here
    Ok(sim_score) // Replace with actual result
}